### PR TITLE
add docs for CI components

### PR DESCRIPTION
**Pull request by eslutsky**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 # forkliftci
-Scripts and tools and github-actions for creating and deploying forklift cluster infrastructure and running integration tests against it.
+This repo contains collection of scripts,tools and gh actions for creating and deploying forklift cluster  and running integration e2e tests against it.
```
Comment (ahadas) [inline README.md:2]:
```suggestion
This repository contains a collection of scripts, tools and GitHub actions for creating and deploying forklift cluster and running e2e integration tests against it.
```

File: README.md [hunk 2]
```diff
@@ -1,5 +1,8 @@
...
 
 * How to [install forklift on _kind_](INSTALL_FORKLIFT_ON_KIND.md).
 * How to [build kubev2v-forklift and install it on _kind_](BUILD_AND_INSTALL_FORKLIFT_ON_KIND.md)
+* migration source  [providers](cluster/providers/README.md)
```
Comment (ahadas) [inline README.md:6]:
```suggestion
* Define migration source [providers](cluster/providers/README.md)
```

File: README.md [hunk 3]
```diff
@@ -1,5 +1,8 @@
...
+* CI e2e jobs and gh [actions](ci/README.md)
```
Comment (ahadas) [inline README.md:7]:
```suggestion
* Execute CI e2e jobs and GitHub [actions](ci/README.md) 
```

File: README.md [hunk 4]
```diff
@@ -1,5 +1,8 @@
 # forkliftci
-Scripts and tools and github-actions for creating and deploying forklift cluster infrastructure and running integration tests against it.
+This repo contains collection of scripts,tools and gh actions for creating and deploying forklift cluster  and running integration e2e tests against it.
 
 * How to [install forklift on _kind_](INSTALL_FORKLIFT_ON_KIND.md).
 * How to [build kubev2v-forklift and install it on _kind_](BUILD_AND_INSTALL_FORKLIFT_ON_KIND.md)
+* migration source  [providers](cluster/providers/README.md)
+* CI e2e jobs and gh [actions](ci/README.md) 
+* okd sno deployment ansible [playbook](cluster/okd-on-ovirt/README.md).
```
Comment (ahadas) [inline README.md:8]:
```suggestion
* OKD single-node-OpenShift (SNO) deployment using Ansible [playbook](cluster/okd-on-ovirt/README.md).
```

File: ci/README.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
+
+# Forklift CI action collection
+This section include reusuable standalone actions that can be used externally,  consumed from the kubev2v/forklift repo (can be any other GH repo).
```
Comment (ahadas) [inline ci/README.md:3]:
```suggestion
This section includes reusable standalone actions that can be used externally, consumed from the kubev2v/forklift repo (can be any other GH repo).
```

File: ci/README.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+## build-and-setup 
+Build and setup the forkliftci env with kind and prepare the env for e2e testing.
+- install kind (k8s) with a local image docker registry
```
Comment (ahadas) [inline ci/README.md:7]:
```suggestion
- Install kind (k8s) with a local docker image registry
```
please change the other parts below to start with capital letters

File: ci/README.md [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+- deploy kubevirt using manifests.
+- build forklift from source using bazel and push to the local registry.
+- deploy forklift from the local registry.
+- install source provider (openstack/ovirt/vsphere).
+
+## prepare-ansible-secrets
+the secrets stored in gh forklift and forkliftci settings.
```
Comment (ahadas) [inline ci/README.md:14]:
```suggestion
The secrets are stored in GitHub settings of the kubev2v/forklift and kubev2v/forkliftci repositories.
```

File: ci/README.md [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+they are stored in base64 and provide configuration 
+for other actions, used by create-self-runner.
+
+
+## create-self-runner
+- provision Self hosted runner from a fedora template.
```
Comment (ahadas) [inline ci/README.md:20]:
```suggestion
- Provisions a self-hosted runner from a Fedora template.
```

File: ci/README.md [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+- join as GH hosted runner.
+
+## run-suite
+Wrapper for running a forklift e2e testing suite,
+all the suites
+
+## deploy-okd
+Use ansible to deploy OKD SNO instance on top of oVirt for e2e testing,refer to this document for the deployment flow.
```
Comment (ahadas) [inline ci/README.md:28]:
```suggestion
Use Ansible to deploy OKD SNO instance on top of oVirt for e2e testing, refer to this document for the deployment flow.
```

File: ci/README.md [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+
+## save-artifacts
+Collect  K8s logs that are relevant for the forklift components. Usefull for troubleshooting e2e migration failures , store them under gh-project job.
```
Comment (ahadas) [inline ci/README.md:31]:
```suggestion
Collect  K8s logs that are relevant for the forklift components. Useful for troubleshooting e2e migration failures, store them under gh-project job.
```

File: cluster/providers/README.md [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+`providers/install-provider.sh <provider-name>`
+
+when `<provider-name>` can be either of:
+- openstack
+- ovirt
+- vmware
+
+
+## openstack 
+containerised [packstack](https://github.com/kubev2v/packstack-img)  deployment  that is tailor made for forklift  migrations.
```
Comment (ahadas) [inline cluster/providers/README.md:15]:
```suggestion
containerised [packstack](https://github.com/kubev2v/packstack-img)  deployment  that is tailored for forklift migrations.
```

File: cluster/providers/README.md [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+    | KEYSTONE_USE_SSL | deploy  Keystone with SSL and self signed certs  | no |
+
+
+## ovirt 
+[fakeovirt](https://github.com/kubev2v/fakeovirt) and [ovirt-imageio](https://github.com/kubev2v/ovirt-imageio-server) deployments :
+- Environment variables:
+
+    | Name | Description | Required | 
+    | ----------- | ----------- | ---  |
+    | NFS_IP_ADDRESS |  NFS server  address used by cinder | yes |
```
Comment (ahadas) [inline cluster/providers/README.md:33]: cinder?

File: cluster/providers/README.md [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+[fakeovirt](https://github.com/kubev2v/fakeovirt) and [ovirt-imageio](https://github.com/kubev2v/ovirt-imageio-server) deployments :
+- Environment variables:
+
+    | Name | Description | Required | 
+    | ----------- | ----------- | ---  |
+    | NFS_IP_ADDRESS |  NFS server  address used by cinder | yes | 
+    | NFS_SHARE |  NFS export path. | yes |
+
+
+## vmware
```
Comment (ahadas) [inline cluster/providers/README.md:37]:
```suggestion
## VMware
```

File: cluster/providers/README.md [hunk 14]
```diff
@@ -0,0 +1,39 @@
...
+vmware/vcsim  deployment  and vsphere-provider with stub vddkInitImage [image](../../stub-images/vddk-test-vmdk/BUILD.bazel).
```
Comment (ahadas) [inline cluster/providers/README.md:38]:
```suggestion
VMware/vcsim deployment and vsphere-provider with stub vddkInitImage [image](../../stub-images/vddk-test-vmdk/BUILD.bazel).
```
we can also add a link to https://github.com/vmware/govmomi/tree/main/vcsim

File: cluster/providers/README.md (hunk 12 above)
Comment (eslutsky) [reply cluster/providers/README.md:33]: Yes we use NFS as [backend](https://github.com/kubev2v/packstack-img/blob/main/files/packstack.answer#L503C1-L503C1) for cinder . 

it is injected to the cinder nfs_shares mounts here https://github.com/kubev2v/forkliftci/blob/main/cluster/providers/openstack/utils.sh#L277 ,
also it used by the OSP volume populator which use csi-driver-nfs   in the backend.

Comment (ahadas) [reply cluster/providers/README.md:33]: ack

**Review by ahadas — APPROVED** [review]

**Pull request merged**